### PR TITLE
feat(cli): promote allow Integration update

### LINKED ISSUE #3673: Allow continuous deployment for `kamel promote`
When we run any `kamel promote` action, there is a check performed to verify that the Integration on destination is not already present:
```
Error: integrations.camel.apache.org "my-rest-jvm" already exists
```
To workaround this we are required to stop the running integration and later run the promote command. I think we need to include a flag (ie, `--continuous`) to skip that check and let the Integration to be refreshed accordingly. Or alternatively, we can skip that check, allowing for an upgrade.

### LINKED ISSUE #3676: Add an info after `kamel promote`
As discussed in https://github.com/apache/camel-k/pull/3674 we should include a short information when promoting an Integration to let the user know the integration was created/updated on the target namespace.

### PR DESCRIPTION
**Pull request by squakez**
Closes #3673
Closes #3676

**Release Note**
```release-note
feat(cli): promote allow Integration update
```

### CODE REVIEW

File: e2e/namespace/install/cli/promote_test.go [hunk 1]
```diff
@@ -107,6 +107,21 @@ func TestKamelCLIPromote(t *testing.T) {
 				Expect(IntegrationPodImage(nsProd, "promote-route")()).Should(Equal(IntegrationPodImage(nsDev, "promote-route")()))
 			})
 
+			t.Run("plain integration promotion update", func(t *testing.T) {
+				// Update the configmap
+				var cmData = make(map[string]string)
+				cmData["my-configmap-key"] = "I am production, but I was updated!"
+				UpdatePlainTextConfigmap(nsProd, "my-cm", cmData)
+
+				Expect(Kamel("promote", "-n", nsDev, "promote-route", "--to", nsProd).Execute()).To(Succeed())
```
Comment (tadayosi) [inline e2e/namespace/install/cli/promote_test.go:116]: So we don't introduce new flag e.g. `--continous` and just let promote always update the integration?  Is it safe in every case?  Shouldn't we at least show some warning message when it's going to overwrite it?

**Review by tadayosi — APPROVED** [review]
One comment but otherwise looks good.

File: e2e/namespace/install/cli/promote_test.go (hunk 1 above)
Comment (squakez) [reply e2e/namespace/install/cli/promote_test.go:116]: Yeah, finally I opted for this solution. When a user is doing a `promote` probably knows what he is doing (similar when he `run` an integration that will update any previous one). Definitely, we should add some output to tell the user either the Integration was created or updated (such as when we `run` them).

**Comment by tadayosi** [comment]
The test is failing:
```
❌ TestKamelCLIPromote (2m40.82s)
✅ TestKamelCLIPromote/kamelet_binding_dev (15.1s)
✅ TestKamelCLIPromote/kamelet_integration_dev (19.65s)
✅ TestKamelCLIPromote/no_configmap_in_destination (200ms)
✅ TestKamelCLIPromote/no_secret_in_destination (160ms)
✅ TestKamelCLIPromote/plain_integration_dev (1m18.36s)
✅ TestKamelCLIPromote/plain_integration_promotion (9.54s)
❌ TestKamelCLIPromote/plain_integration_promotion_update (150ms)
```

**Comment by squakez** [comment]
Yeah, sorry, I was on PTO. Already identified the problem and working to its resolution.

**Pull request merged**